components/com_users/Model/UsersModel.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * @package     Joomla.Site
+ * @subpackage  com_users
+ *
+ * @copyright   Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+namespace Joomla\Component\Users\Site\Model;
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\MVC\Model\ListModel;
+
+/**
+ * Model class for list of users.
+ *
+ * @since  4.0
+ */
+class UsersModel extends ListModel
+{
+
+	/**
+	 * Method to auto-populate the model state.
+	 *
+	 * This method should only be called once per instantiation and is designed
+	 * to be called on the first call to the getState() method unless the model
+	 * configuration flag to ignore the request is set.
+	 *
+	 * Note. Calling getState in this method will result in recursion.
+	 *
+	 * @param   string  $ordering   An optional ordering field.
+	 * @param   string  $direction  An optional direction (asc|desc).
+	 *
+	 * @return  void
+	 *
+	 * @since   12.2
+	 */
+	protected function populateState($ordering = 'ordering', $direction = 'ASC')
+	{
+		$app = \JFactory::getApplication();
+
+		// List state information
+		$groupId = $app->input->get('id');
+		$this->setState('user.group', $groupId);
+
+		parent::populateState($ordering = 'ordering', $direction = 'ASC');
+	}
+
+	/**
+	 * Get the list of items.
+	 *
+	 * @return \JDatabaseQuery|\Joomla\Database\DatabaseQuery
+	 */
+	protected function getListQuery()
+	{
+		$app = \JFactory::getApplication();
```

Comment:
You can use here an existing function from Access:
\Joomla\CMS\Access\Access::getUsersByGroup((int)$this->getState('user.group'));

Reply:
I looked into how to use this, but what I do in this function is to also filter by access, and if I use Access::getUsersByGroup I only get the users ids and then I have to get all the data for the users and then to filter by access and for that I have to make multiple queries in db. I did all that in one query which I think is more efficient.